### PR TITLE
[BD-46] feat: add message about file restrictions to default state of `Dropzone`

### PR DESCRIPTION
**Pull request by viktorrusakov**
## Description

Auto-generate message about file restrictions in `Dropzone` component based on `accept`, `minSize` and `maxSize` props.

### Deploy Preview

https://deploy-preview-1470--paragon-openedx.netlify.app/components/dropzone/

## Merge Checklist

* [x] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [x] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [x] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [x] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [x] Were your changes tested in the `example` app?
* [x] Is there adequate test coverage for your changes?
* [x] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @viktorrusakov!


When this pull request is ready, tag your edX technical lead.

File: src/Dropzone/DefaultContent.jsx [hunk 1]
```diff
@@ -1,21 +1,79 @@
...
+      const allTypes = getTypesString(accept);
+      const lastTypeSeparatorLocation = allTypes.lastIndexOf(',');
+      fileTypePart = intl.formatMessage(messages.fileTypeRestriction, {
+        count: lastTypeSeparatorLocation === -1 ? 1 : 2,
+        firstPart: lastTypeSeparatorLocation === -1 ? allTypes : allTypes.slice(0, lastTypeSeparatorLocation),
+        secondPart: lastTypeSeparatorLocation !== -1 && allTypes.slice(lastTypeSeparatorLocation + 1),
+      });
+    }
+
+    if (minSize && maxSize && Number.isFinite(maxSize)) {
```
Comment (viktorrusakov) [inline src/Dropzone/DefaultContent.jsx:26]: maxSize is `Infinity` by default, hence the `isFinite` check

File: src/Dropzone/DefaultContent.jsx [hunk 2]
```diff
@@ -1,21 +1,79 @@
...
+      </div>
+      <p>
+        <FormattedMessage
+          id="pgn.Dropzone.DefaultContent.label"
+          description="A text that appears as a label for input of Dropzone component."
+          defaultMessage="Drag and drop your file here or click to upload."
+        />
+      </p>
+      {[accept, minSize, maxSize].some(value => value) && (
+        <p className="x-small text-gray-500">
```
[the remark this answers is not in the document]
Comment (viktorrusakov) [reply src/Dropzone/DefaultContent.jsx]: I always forget about this one 😅 
thanks, updated!

**Review by adamstankiewicz — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@viktorrusakov 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/1470&entry.1484655318=https://github.com/viktorrusakov&entry.752974735=2022-07-13+06:47&entry.1917517419=2022-07-29+10:30&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**